classes/local/updatemetadata_form.php

```diff
@@ -0,0 +1,150 @@
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Upload video form.
+ *
+ * @package    block_opencast
+ * @copyright  2019 Farbod Zamani, ELAN e.V.
+ * @author     Farbod Zamani
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+namespace block_opencast\local;
+
+defined('MOODLE_INTERNAL') || die();
+
+global $CFG;
+
+require_once($CFG->dirroot . '/lib/formslib.php');
+
+class updatemetadata_form extends \moodleform {
+
+    public function definition() {
+
+        $mform = $this->_form;
+
+        $metadata = $this->_customdata['metadata'];
+        
+        unset($metadata[array_search('isPartOf', array_column($metadata,'id'))]);
+        $metadata = array_values($metadata);
+
+        $date = $metadata[array_search('date', array_column($metadata,'type'))];
+        $time = $metadata[array_search('time', array_column($metadata,'type'))];
+        $date_time = strtotime($date->value . ' '  . $time->value);
+
+        unset($metadata[array_search('startTime', array_column($metadata,'id'))]);
+        $metadata = array_values($metadata);
+
+
+        $languages = $this->_customdata['languages'] ? json_decode($this->_customdata['languages']->param_json, true) : [
+            "" => "No option selected",
+            "slv" => ucwords(strtolower("SLOVENIAN")),
+            "por" => ucwords(strtolower("PORTUGESE")),
+            "roh" => ucwords(strtolower("ROMANSH")),
+            "ara" => ucwords(strtolower("ARABIC")),
+            "pol" => ucwords(strtolower("POLISH")),
+            "ita" => ucwords(strtolower("ITALIAN")),
+            "zho" => ucwords(strtolower("CHINESE")),
+            "fin" => ucwords(strtolower("FINNISH")),
+            "dan" => ucwords(strtolower("DANISH")),
+            "ukr" => ucwords(strtolower("UKRAINIAN")),
+            "fra" => ucwords(strtolower("FRENCH")),
+            "spa" => ucwords(strtolower("SPANISH")),
+            "gsw" => ucwords(strtolower("SWISS GERMAN")),
+            "nor" => ucwords(strtolower("NORWEGIAN")),
+            "rus" => ucwords(strtolower("RUSSIAN")),
+            "jpx" => ucwords(strtolower("JAPANESE")),
+            "nld" => ucwords(strtolower("DUTCH")),
+            "tur" => ucwords(strtolower("TURKISH")),
+            "hin" => ucwords(strtolower("HINDI")),
+            "swa" => ucwords(strtolower("SWEDISH")),
+            "eng" => ucwords(strtolower("ENGLISH")),
+            "deu" => ucwords(strtolower("GERMAN"))
+        ];
```

Comment:
Why use ucwords and strtolower? Barely improves readability and adds functioncalls